### PR TITLE
Remove some minor unnecessary CMake options

### PR DESCRIPTION
**Pull request by msimberg**
Removes `HPX_WITH_DEFAULT_TARGETS`, `HPX_WITH_PSEUDO_DEPENDENCIES`, `HPX_WITH_MORE_THAN_64_THREADS`, and `HPX_WITH_NATIVE_TLS`. The last two are definitely unnecessary. `HPX_WITH_PSEUDO_DEPENDENCIES` is always off on Windows, on otherwise, so I've just made it an internal variable. `HPX_WITH_DEFAULT_TARGETS` I could see someone finding useful so please speak up if you think it should be kept around.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -293,17 +293,9 @@ if(HPX_WITH_DOCUMENTATION)
 endif()
 
 if(MSVC)
-  hpx_option(
-    HPX_WITH_PSEUDO_DEPENDENCIES BOOL
-    "Force creating pseudo targets and pseudo dependencies (default OFF)." OFF
-    CATEGORY "Build Targets"
-  )
+  set(HPX_WITH_PSEUDO_DEPENDENCIES ON)
```
[the remark this answers is not in the document]
Comment (msimberg) [reply CMakeLists.txt:296]: Indeed, thank you!

[the remark this answers is not in the document]
Comment (hkaiser) [reply CMakeLists.txt]: Actually, this is a Windows limitation, and not related to MSVC.

[the remark this answers is not in the document]
Comment (msimberg) [reply CMakeLists.txt]: Is `WIN32` the correct way to identify it in that case?

[the remark this answers is not in the document]
Comment (hkaiser) [reply CMakeLists.txt]: Yes, I think so: https://cmake.org/cmake/help/v3.20/variable/WIN32.html

**Pull request merged**